requests-for-comments/2023-03-01-Temporal-RFC.md

```diff
@@ -0,0 +1,239 @@
+
+What is a compounding factor here is that this isn’t just a plugin server issue but an issue that is felt across the product: Async migrations, billing, and PoE are all tasks that fit a workflow engine really well. There will be other tasks that we need to run in the future that will need a solid framework for executing workflows as well. This allows us to focus on these and not something that is outside of our expertise and product space.
+
+# Temporal
+
+Temporal is a workflow engine that provides a way for developers to build and manage distributed applications. It is designed to help developers focus on their business logic by abstracting away the details of failure modes, retry logic, and timeouts. Temporal allows developers to build workflows that describe complex processes as a set of simple, easy-to-understand steps. It provides a programmable interface that enables developers to write the code that defines their workflows and the logic that controls how they execute.
+
+Temporal is used by a number of companies, including Uber, DoorDash, Instacart, HashiCorp, and Coinbase. It has become a popular choice among developers because it provides a way to manage complex distributed systems without having to worry about the underlying infrastructure. Temporal allows developers to build workflows that are resilient, scalable, and fault-tolerant, which is essential for building applications that can handle large volumes of data and traffic.
+
+One of the key features of Temporal is its ability to provide visibility into the status of workflows. It provides a dashboard that allows developers to monitor the progress of their workflows and to see where errors are occurring. This helps developers to quickly identify and fix problems, which is essential for ensuring that workflows are running smoothly.
+
+Overall, Temporal is a powerful tool that enables developers to build complex distributed applications and workflows. It provides a way to manage the complexity of distributed systems and to build applications that are scalable, resilient, and fault-tolerant.
+
+[What is temporal?](https://docs.temporal.io/temporal)
+
+[How does temporal work?](https://temporal.io/how-temporal-works)
+
+### Why is it a good fit?
+
+Temporal is an especially good fit because it provides things that are genuinely hard to build and maintain ourselves:
+
+- Retry logic per Activity and Workflow
+- Timeout logic per activity and workflow
+- Exactly-once semantics (!!)
+- Go, [Python](https://docs.temporal.io/kb/python-sandbox-environment), and Typescript libraries that support deterministic code execution and retries (!!)
+- Composability of Activities and Workflows
+- Dashboard for observing Workflow status
+
+Finally, Temporal is a good fit because we are a small team. We should be focusing on the business of building our product. Nothing about what we are building that would be run on Temporal is novel enough to justify us to invest in a competing runtime.
```

Comment:
Agreed, these systems can be intellectually satisfying to write, but extremely hard to debug at 3AM.
Given our upcoming scaling challenges, I'm happy to go with boring but reliable.